tests/foreman/api/test_user.py

```diff
@@ -675,3 +677,264 @@ def test_positive_info_users_ssh_key(self):
+            attr_mail=LDAP_ATTR['mail'],
+            name=gen_string('alpha'),
+            host=settings.ipa.hostname_ipa,
+            tls=False,
+            port='389',
+            location=[cls.loc],
+            organization=[cls.org],
+        ).create()
+
+    def tearDown(self):
+        for user in entities.User().search(query={'search': u'login=' + self.username}):
+            user.delete()
+        super(FreeIPAUserTestCase, self).tearDown()
+
+    @tier3
+    @skip_if_not_set('ipa')
+    def test_positive_ipa_basic_no_roles(self):
+        """Login with LDAP Auth- FreeIPA for user with no roles/rights
+
+        :id: 901a241d-aa76-4562-ab1a-a752e6fb7ed5
+
+        :setup: assure properly functioning FreeIPA server for authentication
+
+        :steps: Login to server with an FreeIPA user.
+
+        :expectedresults: Log in to foreman successfully but cannot access entities.
+
+        :CaseLevel: System
+        """
+        sc = ServerConfig(
+            auth=(self.username, self.ldap_user_passwd),
+            url=self.sat_url,
+            verify=False
+        )
+        with self.assertRaises(HTTPError):
+            entities.Architecture(sc).search()
+
+    @tier3
+    @upgrade
+    @skip_if_not_set('ipa')
+    def test_positive_access_entities_from_ipa_org_admin(self):
+        """LDAP FreeIPA User can access resources within its taxonomies if assigned
+        role has permission for same taxonomies
+
+        :id: acc48330-fc84-4970-b722-9b45a3116eed
+
+        :steps:
+
+            1. Create Org Admin and assign taxonomies to it
+            2. Create FreeIPA user with same taxonomies as role above
+            3. Assign Org Admin role to user above
+            4. Login with FreeIPA user and attempt to access resources
+
+        :expectedresults: FreeIPA User should be able to access all the resources
+            and permissions in taxonomies selected in Org Admin role
+
+        :CaseLevel: System
+        """
+        name = gen_string('alpha')
+        default_org_admin = entities.Role().search(
+            query={'search': u'name="Organization admin"'})
+        org_admin = entities.Role(id=default_org_admin[0].id).clone(
+            data={
+                'role': {
+                    'name': name,
+                    'organization': self.org.name,
+                    'location': self.loc.name
+                }
+            }
+        )
+        sc = ServerConfig(
+            auth=(self.username, self.ldap_user_passwd),
+            url=self.sat_url,
+            verify=False
+        )
+        with self.assertRaises(HTTPError):
+            entities.Architecture(sc).search()
+        user = entities.User().search(query={'search': u'login=' + self.username})[0]
+        user.role = [entities.Role(id=org_admin['id']).read()]
+        user.update(['role'])
+        with self.assertNotRaises(HTTPError):
+            entities.Architecture(sc).search()
```

Comment:
same here and maybe we can make that list static, so no copy paste same code twice